web/src/components/molecules/Common/ResizableProTable/index.tsx

```diff
@@ -0,0 +1,87 @@
+import styled from "@emotion/styled";
+import { useEffect, useState } from "react";
+
+import ProTable, {
+  ProColumns,
+  OptionConfig,
+  ListToolBarProps,
+} from "@reearth-cms/components/atoms/ProTable";
+import { ResizableTitle } from "@reearth-cms/components/molecules/Common/ResizableProTable/resizble";
+import type { ResizeCallbackData } from "@reearth-cms/components/molecules/Common/ResizableProTable/resizble";
+
+export type Props = {
+  className?: string;
+  dataSource?: any[];
+  proColumns?: ProColumns<any, "text">[];
+  loading: boolean;
+  options?: false | OptionConfig | undefined;
+  toolbar?: ListToolBarProps | undefined;
+};
+
+const ResizableProTable: React.FC<Props> = ({
+  dataSource,
+  proColumns,
+  loading,
+  options,
+  toolbar,
+}) => {
+  const [columns, setColumns] = useState<ProColumns<any, "text">[]>([]);
+
+  useEffect(() => {
+    if (Array.isArray(proColumns)) {
+      setColumns(proColumns);
+    }
+  }, [proColumns, setColumns]);
+
+  const handleResize =
+    (index: number) =>
+    (_: React.SyntheticEvent<Element>, { size }: ResizeCallbackData) => {
+      const newColumns = [...columns];
+      newColumns[index] = {
+        ...newColumns[index],
+        width: size.width,
+      };
+      setColumns(newColumns);
+    };
+
+  const mergeColumns: ProColumns<any, "text">[] = columns?.map((col, index) => ({
+    ...col,
+    onHeaderCell: column => ({
+      minWidth: (column as ProColumns<any, "text"> & { minWidth: number }).minWidth,
+      width: (column as ProColumns<any, "text">).width,
```

Comment:
Do you need minWidth if you are setting width? Typically if you set width it'll be that size always

Reply:
@KaWaite , `width` is a changeable property depending on the dragging, and antD depends on it to set the `width` so I need the minWidth to be a fixed value and prevent the dragging from being less than its value.